trunk/source/PEAtoBoogie/src/de/uni_freiburg/informatik/ultimate/pea2boogie/InvariantResultTransformer.java

```diff
@@ -0,0 +1,50 @@
+package de.uni_freiburg.informatik.ultimate.pea2boogie;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import de.uni_freiburg.informatik.ultimate.boogie.ast.BinaryExpression;
+import de.uni_freiburg.informatik.ultimate.boogie.ast.Expression;
+import de.uni_freiburg.informatik.ultimate.boogie.ast.IdentifierExpression;
+import de.uni_freiburg.informatik.ultimate.boogie.ast.UnaryExpression;
+
+public class InvariantResultTransformer {
+
+	private InvariantResultTransformer() {
+		// Utility
+		// This class could serve for more purposes in the future and also allow
+		// instances if needed but right now it's really just to have the extraction
+		// outside of other classes where it doesn't belong. In the future perhaps it could
+		// post-process for several reqcheck properties such as e.g., vacuity too.
+	}
+
+	public static Set<String> extractRedundancySet(final Expression invariant) {
+		final var redSet = new HashSet<String>();
```

Comment:
It might be better to use `LinkedHashSet<String>` here to prevent flaky results when iterating over the found redundancies later on.

Reply:
Is this purely about also keeping track of the insertion order for iteration? For our purposes it doesn't really matter, but if you think it can be useful / is a better and cleaner way to approach it I can still change it. Since I'm using the BoogieIdExtractor class now, it is still returned as a HashSet.

Reply:
Yes, purely about the order when converting to string. It might then be helpful to convert the set to a list and sort it before printing it, s.t. users dont see non-deterministic output.

Reply:
I am using a list within the results now that is also sorted for more deterministic printing